File: 07_multiply_polynomials.py
def multiplyPolynomials(poly1, poly2):
    result = [0] * (len(poly1) + len(poly2) - 1)
    for i in range(len(poly1)):
        for j in range(len(poly2)):
            result[i+j] += poly1[i] * poly2[j]
    return result

print("Polynomial Multiplication:", multiplyPolynomials([1, 2], [1, 1]))

File: 07_multiply_polynomials.c
#include <stdio.h>

int main() {
    int n, m;
    printf("Enter degree of 1st polynomial: ");
    scanf("%d", &n);
    int poly1[n+1];
    printf("Enter coefficients:\n");
    for(int i=0; i<=n; i++) scanf("%d", &poly1[i]);

    printf("Enter degree of 2nd polynomial: ");
    scanf("%d", &m);
    int poly2[m+1];
    printf("Enter coefficients:\n");
    for(int i=0; i<=m; i++) scanf("%d", &poly2[i]);

    int prod[n+m+1];
    for(int i=0; i<=n+m; i++) prod[i] = 0;

    for(int i=0; i<=n; i++)
        for(int j=0; j<=m; j++)
            prod[i+j] += poly1[i] * poly2[j];

    printf("Product polynomial: ");
    for(int i=n+m; i>=0; i--) {
        if(prod[i] != 0)
            printf("%d*x^%d ", prod[i], i);
    }
    return 0;
}